subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/OutgoingVariantsReportTask.java

```diff
@@ -15,323 +15,49 @@
-import org.gradle.api.attributes.AttributeContainer;
-import org.gradle.api.capabilities.Capability;
-import org.gradle.api.internal.artifacts.ProjectBackedModule;
-import org.gradle.api.internal.artifacts.configurations.ConfigurationInternal;
-import org.gradle.api.internal.attributes.AttributeContainerInternal;
-import org.gradle.api.internal.file.FileResolver;
-import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.Optional;
-import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.diagnostics.internal.configurations.spec.AbstractConfigurationReportSpec;
+import org.gradle.api.tasks.diagnostics.internal.configurations.spec.OutgoingVariantsSpec;
 import org.gradle.api.tasks.options.Option;
-import org.gradle.internal.logging.text.StyledTextOutput;
-import org.gradle.internal.logging.text.StyledTextOutputFactory;
 import org.gradle.work.DisableCachingByDefault;
 
-import javax.annotation.Nullable;
-import javax.inject.Inject;
-import java.io.File;
-import java.util.Collection;
-import java.util.Comparator;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
 /**
  * A task which reports the outgoing variants of a project on the command line.
+ *
  * This is useful for listing what a project produces in terms of variants and
  * what artifacts are attached to each variant.
+ *
  * Variants, in this context, must be understood as "things produced by a project
  * which can safely be consumed by another project".
  *
  * @since 6.0
  */
-@DisableCachingByDefault(because = "Produces only non-cacheable console output")
-public class OutgoingVariantsReportTask extends DefaultTask {
-    private final Property<String> variantSpec = getProject().getObjects().property(String.class);
-    private final Property<Boolean> showAll = getProject().getObjects().property(Boolean.class).convention(false);
-
+@DisableCachingByDefault(because = "Produces only non-cacheable console output by examining configurations at execution time")
+public abstract class OutgoingVariantsReportTask extends AbstractConfigurationReportTask {
+    /**
+     * Limits the report to a single variant.
+     *
+     * @return property holding name of the variant to report
+     */
     @Input
     @Optional
-    @Option(option = "variant", description = "The variant name")
-    Property<String> getVariantName() {
-        return variantSpec;
-    }
-
+    @Option(option = "variant", description = "The name of a single variant to report")
+    public abstract Property<String> getVariantName();
+
+    /**
+     * Shows all variants, including legacy and deprecated configurations.
+     *
+     * @return property holding the flag to show all variants
+     */
     @Input
     @Optional
     @Option(option = "all", description = "Shows all variants, including legacy and deprecated configurations")
```

Comment:
```suggestion
    @Option(option = "all", description = "Shows all variants, including from legacy and deprecated configurations")
```
Perhaps this or some other similar change, since otherwise it creates more "variant" and "configuration" confusion.

Reply:
In this case, I think I prefer the original.  Variants FROM legacy and deprecated configurations seems to say that variants are not a part of the l and d confs.  It is a little confusing either way, but I think when running reports with additional options we have to trust that the user understands the equivalence.